### ISSUE TITLE
Support Angular 10

### ISSUE THREAD

**Issue by TrevorKarjanis** [plotly/angular-plotly.js#120]
Angular 10 has been [released](https://blog.angular.io/version-10-of-angular-now-available-78960babd41)! Due to upgrades in other dependencies, it would be helpful if angular-plotly.js upgraded as well.

I went ahead and followed the [update guide](https://update.angular.io/#9.0:10.0) for angular-plotly.js, validated the tests pass, and brought it in to a 10 application for a short test. Please let me know if you would like a PR. Those changes can be seen [here](https://github.com/plotly/angular-plotly.js/compare/master...TrevorKarjanis:upgrade-angular). Here are some items of note.
- Angular 10 drops support for IE 9 and 10.
- As per the update guide, "The Angular Package Format has been updated to remove esm5 and fesm5 formats. These are no longer distributed in our npm packages."
- As per the update guide, "Remove any es5BrowserSupport flags in your angular.json and set your target to es2018 in your tsconfig.json. Angular now uses your browserslist to determine if an ES5 build is needed." ng update did not create a browserlist file, so I created one that supports IE11+.
- From the library docs, "Furthermore, the internal Ivy instructions are not yet stable, which can potentially break consumers using a different Angular version from the one used to build the library." It's probably recommend to include this dependency in version 10 apps.

**Comment by andrefarzat** [comment]
Hello @TrevorKarjanis , we will release a compatible version with angular 10.x soon. It will be in a different major release

**Comment by fourgates** [comment]
This may be an unnecessary comment as a subsequent major release may fix this, but I tried to upgrade my project to Angular 10 from Angular 9 this weekend. I am now getting the following error:

```
Invalid PlotlyJS object. Please check https://github.com/plotly/angular-plotly.js#quick-start to see how to add PlotlyJS to your project.
```

Here is my module:

```
import * as PlotlyJS from 'plotly.js/dist/plotly.js';
import { PlotlyModule } from 'angular-plotly.js';
PlotlyModule.plotlyjs = PlotlyJS;
console.log('PlotlyModule.plotlyjs', PlotlyModule.plotlyjs);
console.log('PlotlyModule_1.plotlyjs !== undefined', PlotlyModule.plotlyjs !== undefined);
console.log('typeof PlotlyModule_1.plotlyjs.plot === function', typeof PlotlyModule.plotlyjs.plot === 'function');
console.log('typeof PlotlyModule.plotlyjs.plot', typeof PlotlyModule.plotlyjs.plot);
```

Note we had followed these instructions to a T when we upgraded to Angular 9 and it was working fine. I added some logging to try to diagnose:

```
main.js:30980 PlotlyModule.plotlyjs: 
ModuleSymbol(Symbol.toStringTag): "Module"__esModule: true__proto__: Objectconstructor: ƒ Object()hasOwnProperty: ƒ hasOwnProperty()isPrototypeOf: ƒ isPrototypeOf()propertyIsEnumerable: ƒ propertyIsEnumerable()toLocaleString: ƒ toLocaleString()toString: ƒ ()valueOf: ƒ valueOf()__defineGetter__: ƒ __defineGetter__()__defineSetter__: ƒ __defineSetter__()__lookupGetter__: ƒ __lookupGetter__()__lookupSetter__: ƒ __lookupSetter__()get __proto__: ƒ __proto__()set __proto__: ƒ __proto__()

main.js:30981 PlotlyModule_1.plotlyjs !== undefined true
main.js:30982 typeof PlotlyModule_1.plotlyjs.plot === function false
main.js:30983 typeof PlotlyModule.plotlyjs.plot undefined
```

Also, note that I had to apply this "hack" to the plotly.js dist file to fix an issue with importing plotly:
```
TypeError: Cannot read property 'document' of undefined
    at plotly.js:24924
```

https://github.com/plotly/plotly.js/issues/3518#issuecomment-517138019

**Comment by andrefarzat** [comment]
@fourgates The main issue here is that `plotly.js` is bundled (concatenated and minified) into the final project code, and the `d3.js` (a `plotly.js` dependency) doesn't bundle up well when it isn't a `es5` module. Once angular 10 has removed this option as default, in for deprecating IE browsers.

Quote:
```This has the side effect of disabling ES5 builds by default for new projects. To enable ES5 builds and differential loading for browsers that require it (such as IE or UC Browser), simply add the browsers you need to support in the .browserslistrc file.```

Source: https://blog.angular.io/version-10-of-angular-now-available-78960babd41
